WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -289,6 +289,83 @@ class PostCoordinator: NSObject {
         })
     }
 
+    func upload(assets: [ExportableAsset], to post: Post, completion: @escaping (Result<Post, SavingError>) -> Void) -> [Media?] {
+        guard mediaCoordinator.uploadMedia(for: post) else {
+            change(post: post, status: .failed) { savedPost in
+                completion(.failure(SavingError.mediaFailure(savedPost)))
+            }
+            return []
+        }
+
+        change(post: post, status: .pushing)
+
+        change(post: post, status: .pushingMedia)
+        // Only observe if we're not already
+        guard !isObserving(post: post) else {
```

Comment:
I love the guard too but I think `if isObserving(post: post) {` it's more clear to understand over double NOT. What do you think?

Reply:
I was trying to keep this the same, particularly in the `prepareToSave` method:

https://github.com/wordpress-mobile/WordPress-iOS/blob/944ac9db29e6be4c5f0519195b3f192bb8b2edc0/WordPress/Classes/Services/PostCoordinator.swift#L136

I think I will come back to this over Hack Week because the observer handling could be simplified and better documented.